example-github/src/main/java/feign/example/github/GitHubExample.java

```diff
@@ -53,4 +64,51 @@ public static void main(String... args) throws InterruptedException {
     String login;
     int contributions;
   }
+
+  static class ClientError {
```

Comment:
I already had a quick shot at it without success. I haven't had the time to go into details though.